### PR TITLE
Adding writer to a flat PostgreSQL table

### PR DESCRIPTION
**Pull request by oscarmartinezrubi**
The writer uses pqputcopydata in binary form to make it as faster as possible (though it still is slow compared to tool las2pg in liblas)

### CODE REVIEW

**Comment by abellgithub** [comment]
Is this a standard way of storing points?  Is it compatible with any other software?

**Comment by oscarmartinezrubi** [comment]
Well, the idea is to use this writer when you want to end up creating pgpointcloud blocks in a different way than what PDAL-pgpointcloud-writer does by default.

Once you import points in a flat table with this new writer, you can create within the DB PC_POINTS:

CREATE TABLE points ( id SERIAL PRIMARY KEY,  pt PCPOINT(1));
INSERT INTO points (pt) SELECT PC_MakePoint(1, ARRAY[x,y,z,intensity]) FROM (SELECT  x,y,z,intensity FROM flat) AS values;

And then you can create PC_PATCHES in any way you want, for example:

CREATE TABLE patches (id SERIAL PRIMARY KEY, pa PCPATCH(1));
INSERT INTO patches (pa) SELECT PC_Patch(pt) FROM points GROUP BY id/10;

So, basically this writer is complementary to the existing pgpointcloud (that is why I put it in the same plugin folder)

**Comment by hobu** [comment]
@oscarmartinezrubi I know you don't have any time to refactor this, but I think we should table this effort until it can be rolled into the existing `writers.pgpointcloud` rather than a separate thing to maintain. Given that people have `las2pg` as an alternative also gives people a way forward.

Thank you for providing the PR. If someone wants this bad enough to pick it up again, they can start improving in the context of the existing driver.
